File: guide1/exercise1/exercise1.cpp
#include <iostream>
#include <utility>

using namespace std;

class Flight {

  private:
    int flightNumber;
    int flightCapacity;
    string origin;
    string destiny;
    string date;

  public:
    Flight(const int flightNumber, const int flightCapacity, string  origin,
          string  destiny, string  date) : flightNumber(flightNumber), flightCapacity(flightCapacity),
          origin(move(origin)), destiny(move(destiny)), date(move(date)) {}

    int getFlightNumber() const { return flightNumber; }
    string getOrigin() const { return origin; }

    bool isAvailable() const {
      return (flightCapacity > 0) ? true : false;
    }

};

class Passenger {

  private:
    string name;
    string rut;
    string email;

  public:
    Passenger(string name, string rut, string email) : name(move(name)),
             rut(move(rut)), email(move(email)) {}

    string getName() const { return name; }
    string getRut() const { return rut; }
    string getEmail() const { return email; }
};

int main() {

  // ill finish this tomorrow
  return 0;
}

File: guide1/exercise1/exercise1.java
public class Main {

    static class Flight {
        private int flightNumber;
        private int flightCapacity;
        private String origin;
        private String destiny;
        private String date;

        public Flight(int flightNumber, int flightCapacity,
                     String origin, String destiny, String date) {

            this.flightNumber = flightNumber;
            this.flightCapacity = flightCapacity;
            this.origin = origin;
            this.destiny = destiny;
            this.date = date;

        }

        public int getFlightNumber() {
            return flightNumber;
        }

        public String getOrigin() {
            return origin;
        }

        public boolean isAvailable() {
            return flightCapacity > 0;
        }
    }

    static class Passenger {
        private String name;
        private String rut;
        private String email;

        public Passenger(String name, String rut, String email) {
            this.name = name;
            this.rut = rut;
            this.email = email;
        }

        public String getName() {
            return name;
        }

        public String getRut() {
            return rut;
        }

        public String getEmail() {
            return email;
        }
    }

    public static void main(String[] args) {

    }
}
